### ISSUE TITLE
Re-shuffle data each iteration

### ISSUE THREAD

**Issue by AlexeyG** [google/flax#166]
Some example (e.g. `nlp_seq`) show clear patterns in their training loss (see TensorBoard).

This is probably because the training data isn't shuffled enough. We should consider adding `shuffle` with `reshuffle_each_iteration=True` to the train IO pipelines of existing examples.

**Comment by jheek** [comment]
`reshuffle_each_iteration` is `False` by default? I guess this is a great example of why implicit random seeds are evil. 

In the past I have also noticed that creating iterators in `tf.data` tends to come with significant overhead so I prefer to create one infinite iterator and use a manual `for step in range(steps_per_epoch)` to loop for an epoch. This also avoids some other issues like partial batches.

**Comment by AlexeyG** [comment]
Just checked, it actually does default to `True`:

```
reshuffle_each_iteration: (Optional.) A boolean, which if true indicates that the dataset should be pseudorandomly reshuffled each time it is iterated over. (Defaults to True.)
```

Perhaps the issue is that the `nlp_seq` example does not shuffle the data?

**Comment by AlexeyG** [comment]
Marking as PR welcome. A successful PR would also include (a link to) new TensorBoard logs verifying the metrics (see `README.md` in `examples/nlp_seq` for expected numbers).

**Comment by gan3sh500** [comment]
It looks like dataset.shuffle was just missing. I've made the change. I'll train on a V100 similar to the README and submit a PR with updated logs and code.

**Comment by avital** [comment]
Awesome @gan3sh500 ! Indeed if you can reproduce the same results on a V100 and share a link to tensorboard.dev we'll merge your PR.

**Comment by gan3sh500** [comment]
Sorry for slow turnaround. I didn't get the V100 but I now have a 2080Ti locally to run and have ran the same batch_size and iterations. The results are 0.23% lower than the one shown in the repo. Please see for yourself. [TB Dev](https://tensorboard.dev/experiment/1kp6O2xbRDWCJCmrmnLBgA) #321

**Comment by marcvanzee** [comment]
It looks like @bohnetbd felt this wasn't worth adding due to the added complexity (given that this is an educational example). So I am closing this.
